pkg/kubelet/cm/devicemanager/checkpoint/checkpoint.go

```diff
@@ -45,14 +49,23 @@ type PodDevicesEntry struct {
 // checkpointData struct is used to store pod to device allocation information
 // in a checkpoint file.
 // TODO: add version control when we need to change checkpoint format.
-type checkpointData struct {
+type checkpointDataV1 struct {
 	PodDeviceEntries  []PodDevicesEntry
 	RegisteredDevices map[string][]string
 }
 
+// checkpointData struct is used to store pod to device allocation information
+// in a checkpoint file.
+// version 2: add AllDevices field
+type checkpointDataV2 struct {
+	PodDeviceEntries  []PodDevicesEntry
+	RegisteredDevices map[string][]string
+	AllDevices        device.ResourceDeviceInstances `json:"AllDevices,omitempty"`
```

Comment:
Why change to V2 instead of adding directly to the original?

Reply:
Mainly to ensure that the check verify sum can be successful after the kubelet is upgraded.